test/functional/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/redefineClasses/rc020.java

```diff
@@ -0,0 +1,165 @@
+		resetClassBytes = new rc020_testIndyAsmGenerator().generateExistingIndyClass();
+
+		/* Redefine class with one indy call. */
+		if (!runTest("testIndyRedefinition (basic test)", new rc020_testIndyAsmGenerator().generateBasicIndyClass(), 1, 0)) {
+			return false;
+		}
+
+		/* Redefine class with one indy call where bsm has an argument. */
+		if (!runTest("testIndyRedefinition (basic test, bsm args)", new rc020_testIndyAsmGenerator().generateBasicIndyClassWithArgs(), 1, 0)) {
+			return false;
+		}
+
+		/* Redefine class with one indy call where bsm has two additional arguments. */
+		if (!runTest("testIndyRedefinition (basic test, bsm two args)", new rc020_testIndyAsmGenerator().generateBasicIndyClassWithTwoArgs(), 1, 0)) {
+			return false;
+		}
+
+		/* Redefine class with one indy call where bsm has an additional double slot argument. */
+		if (!runTest("testIndyRedefinition (basic test, bsm double slot args)", new rc020_testIndyAsmGenerator().generateBasicIndyClassWithDoubleSlotArgs(), 1, 0)) {
+			return false;
+		}
+
+		/* Redefine class with one indy call where bsm has three args with a double slot in the middle */
+		if (!runTest("testIndyRedefinition (basic test, bsm double slot with three args)", new rc020_testIndyAsmGenerator().generateBasicIndyClassWithDoubleSlotWithThreeArgs(), 1, 0)) {
+			return false;
+		}
+
+		/* Redefine class with two indy calls referring to the same BSMs. */
+		if (!runTest("testIndyRedefinition (double indy test, same bsms)", new rc020_testIndyAsmGenerator().generateTwoSameIndyCallClass(), 2, 0)) {
+			return false;
+		}
+
+		/* Redefine class with two indy calls referring to different BSMs. */
+		if (!runTest("testIndyRedefinition (double indy test, diff bsms)", new rc020_testIndyAsmGenerator().generateTwoDiffIndyCallClass(), 2, 0)) {
+			return false;
+		}
+
+		/* Redefine class with two indy calls referring to different BSMs with arguments. */
+		if (!runTest("testIndyRedefinition (double indy test, diff bsms with args)", new rc020_testIndyAsmGenerator().generateTwoDiffIndyCallClassWithArgs(), 2, 0)) {
+			return false;
+		}
+
+		/* Redefine class with three indy calls where two are referring to the same BSM. */
+		if (!runTest("testIndyRedefinition (triple indy test)", new rc020_testIndyAsmGenerator().generateThreeIndyCallClass(), 3, 0)) {
+			return false;
+		}
+
+		/* Redefine class with more than one method with an indy call. */
+		if (!runTest("testIndyRedefinition (multi class test)", new rc020_testIndyAsmGenerator().generateMultiMethodClass(), 1, 0)) {
+		 	return false;
+		}
+
+		/* Redefine class with one indy call where bsm has three args with a double slot in the middle. If the static arguments
+		 * to the BSM change, the output will also change and BSM should be reinvoked. Verify that values are checked correctly.
+		 */
+		if (!runTest("testIndyRedefinition (basic test, bsm double slot with three args parameter change- should fail)", 
```

Comment:
there is one test here that changes the static arguments for the same bsm where the callsite shouldn't be copied. I'll add one where the NAS changes as well for good measure.